src/test/java/core/basesyntax/FruitParserServiceImplTest.java

```diff
@@ -0,0 +1,50 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.ParserService;
+import core.basesyntax.service.impl.FruitParserServiceImpl;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+
+class FruitParserServiceImplTest {
+    private static final String APPLE = "apple";
+    private static final String BANANA = "banana";
+    private static final String INVALID_OPERATION = "x";
+    private final ParserService parserService = new FruitParserServiceImpl();
+
+    @Test
+    void parse_validInput_success() {
+        List<String> commands = List.of("b,banana,20", "s,apple,100");
+        List<FruitTransaction> transactions = parserService.parse(commands);
+
+        assertEquals(2, transactions.size());
+        assertEquals(Operation.BALANCE, transactions.get(0).operation());
+        assertEquals("banana", transactions.get(0).fruit());
+        assertEquals(20, transactions.get(0).quantity());
+        assertEquals(Operation.SUPPLY, transactions.get(1).operation());
+        assertEquals("apple", transactions.get(1).fruit());
+        assertEquals(100, transactions.get(1).quantity());
```

Comment:
Is it possible to create expected objects and compare them? it will make code more clear